docs: migration guide for v1beta1 (#2859)
kyma-project/telemetry-manager: docs/user/migration-guide.md

> # Migrate Your Pipelines from v1alpha1 to v1beta1

Reviewer: Oh! I didn't know that. 
I thought all customers are affected and must take action, that's why we put it on the top navigation level. 
This significantly changes the use case of the migration guide.

Migrate your Telemetry pipeline resources to the stable v1beta1 API version. The v1beta1 API is more consistent, clear, and aligned with industry standards.

This migration involves breaking changes. You must update the **apiVersion**, rename several fields, and, for LogPipeline resources, adjust how you configure namespace selection.

## Context

The migration from v1alpha1 and v1beta1 mostly affects LogPipeline and MetricPipeline resources. For TracePipeline resources, you only have to change the **apiVersion**.

![LogPipeline Migration Changes](./assets/logpipeline-migration.png)
![MetricPipeline Migration Changes](./assets/metricpipeline-migration.png)

In your Telemetry pipeline resources, see if one of the following breaking changes between v1alpha1 and v1beta1 affects you:

| Pipeline                    | v1alpha1 Field                                 | v1beta1 Field                           | Migration Action                                                     |
|-----------------------------|------------------------------------------------|-----------------------------------------|----------------------------------------------------------------------|

> | LogPipeline                 | spec.input.application                         | spec.input.runtime                      | Rename the field.                                                    | LogPipeline                 | spec.output.http.tls.disabled                  | spec.output.http.tls.insecure           | Rename the field.                                                    |

Reviewer: I believe a `|` is missing somewhere, maybe here:

```suggestion
| LogPipeline                 | spec.input.application                         | spec.input.runtime                      | Rename the field.                  |                                  | LogPipeline                 | spec.output.http.tls.disabled                  | spec.output.http.tls.insecure           | Rename the field.                                                    |
```

## Prerequisites

- You have an active Kyma cluster with the Telemetry module added.
- You have one or more Telemetry pipeline resources that use the `telemetry.kyma-project.io/v1alpha1` API.

## Procedure

1. In each of your LogPipeline, MetricPipeline, and TracePipeline YAML files, change the **apiVersion** to `telemetry.kyma-project.io/v1beta1`.

2. For LogPipeline and MetricPipeline resources, find and replace the fields that were renamed in v1beta1.
   1. Update the OTLP input field:
      - **spec.input.otlp.disabled** becomes **spec.input.otlp.enabled**.
      - You must also invert the boolean value (for example, `disabled: false` becomes `enabled: true`).

   2. For LogPipeline resources, also update the following fields:
      - **spec.input.application** becomes **spec.input.runtime**.
      - In the http output, **spec.output.http.tls.disabled** becomes **spec.output.http.tls.insecure**.
      - In the http output, **spec.output.http.tls.skipCertificateValidation** becomes **spec.output.http.tls.insecureSkipVerify**.

3.  For LogPipeline resources, if you want to include system namespaces for application logs, update the system namespace selection.

>    By default, system namespaces are excluded (as in v1alpha1), but v1beta1 removes the **spec.input.application.namespaces.system** field. To include application logs from system namespaces (like `kyma-system`), you must now provide an empty object `({})` for the **namespaces** selector. For details, see [Filter Application Logs by Namespace](https://kyma-project.io/external-content/telemetry-manager/docs/user/filter-and-process/filter-logs.html#filter-application-logs-by-namespace).

Reviewer: Trying to align the empty struct syntax:

```suggestion
   By default, system namespaces are excluded (as in v1alpha1), but v1beta1 removes the **spec.input.application.namespaces.system** field. To include application logs from system namespaces (like `kyma-system`), you must now provide an empty object (`{}`) for the **namespaces** selector. For details, see [Filter Application Logs by Namespace](https://kyma-project.io/external-content/telemetry-manager/docs/user/filter-and-process/filter-logs.html#filter-application-logs-by-namespace).
```

# Migrate Your Pipelines from v1alpha1 to v1beta1

> Telemetry module v1beta1 introduces a number of changes to the Telemetry pipeline API, including some breaking changes.

Reviewer: ```suggestion
Migrate your Telemetry pipeline resources to the stable v1beta1 API version. The v1beta1 API is more consistent, clear, and aligned with industry standards.
```

This guide outlines the key changes and steps required to migrate your Telemetry pipelines from version v1alpha1 to v1beta1. The new version introduces several renamed fields and modifications to improve API functionality and usability.

## Key Changes

The diagram below summarizes the main changes between v1alpha1 and v1beta1 of the Telemetry pipeline API.

![LogPipeline Migration Changes](./assets/logpipeline-migration.png)
![MetricPipeline Migration Changes](./assets/metricpipeline-migration.png)

The following sections provide details on the most important changes in the pipeline API.

> 

Reviewer: ```suggestion

## Prerequisites

-You have an active Kyma cluster with the Telemetry module added.
- You have one or more Telemetry pipeline resources that use the `telemetry.kyma-project.io/v1alpha1` API.

```

# Migrate Telemetry Pipelines to v1beta1

The Telemetry module now uses the stable v1beta1 API for pipeline resources. In some cases, you must update your pipeline manifests to use the new API version.

## Context

If your pipelines are already running in the cluster and you don't store their manifests, you don't have to take any action. The Telemetry module automatically migrates existing your pipeline resources to v1beta1.

> **Tip:** If you're unsure what your existing pipeline looks like in `v1beta1`, you can fetch the converted resource directly from your cluster:
> the converted resource directly from your cluster:
> ```bash
> kubectl get logpipeline <name> -o yaml
> kubectl get metricpipeline <name> -o yaml
> kubectl get tracepipeline <name> -o yaml
> ```

If you manage your Telemetry pipeline resources declaratively, you must manually adjust them; for example, in the following cases:
- Use GitOps, CI/CD, or tools like Argo CD to deploy your pipelines

> - Store pipeline manifests in a Git repository that you re-apply

Reviewer: ```suggestion
- You store pipeline manifests in a Git repository that you re-apply.
```

> # Sync Pipeline Manifests with the New v1beta1 API

Author: At first I thought that the word migration has the implication that every user needs to do actively do this, but in hindsight it doesn't really make sense i guess, i think as long as it is clearly explained in the rest of the page, it should be fine

## Do I Need to Take Action?

**No action required** if your pipelines are already deployed in your cluster.
Existing pipelines are automatically migrated to v1beta1 by the Telemetry module.

**Action required** if you:
- Use GitOps, CI/CD, or tools like Argo CD to deploy your pipelines
- Store pipeline manifests in a Git repository that you re-apply
- Plan to create new pipelines and want to use the v1beta1 API

> **Tip:** If you're unsure how your existing pipeline looks in v1beta1, you can fetch
> the converted resource directly from your cluster:
> ```bash
> kubectl get logpipeline <name> -o yaml
> kubectl get metricpipeline <name> -o yaml
> kubectl get tracepipeline <name> -o yaml
> ```

## Context

This migration involves breaking changes. You must update the **apiVersion**, rename several fields, and, for LogPipeline resources, adjust how you configure namespace selection. The migration from v1alpha1 and v1beta1 mostly affects LogPipeline and MetricPipeline resources. For TracePipeline resources, you only have to change the **apiVersion**.

![LogPipeline Migration Changes](./assets/logpipeline-migration.png)

> ![MetricPipeline Migration Changes](./assets/metricpipeline-migration.png)

Reviewer: Line 30: Maybe users don't have to edit their resources, but the manifests?

"To identify the required updates for your manifests, review the breaking changes in the following table:"
